src/hrflow_connectors/connectors/boards/indeed/actions.py

```diff
@@ -0,0 +1,267 @@
+    )
+    job_search: str = Field(
+        ...,
+        description="Name of the job position we want to search offers in 'fr.indeed.com'",
+    )
+    job_location: str = Field(..., description="Location of the job offers")
+
+    limit: int = Field(
+        15,
+        description=" limit of jobs extracted on page, usually on 'indeed.com', the number of offers per page is 15",
+    )
+    limit_extract: int = Field(1, description=" limit of pages you want to extract")
+
+    @property
+    def url_base(self) -> str:
+        return "https:/{}.indeed.com/".format(self.subdomain)
+
+    def path(self, pagination: int) -> str:
+
+        return "emplois?q={query}&l={location}&limit={limit}&start={start}".format(
+            query=self.job_search,
+            location=self.job_location,
+            limit=self.limit,
+            start=pagination,
+        )
+
+    def pull(self) -> list:
+        """the role of this function is to interact with indeed, click buttons and search offers based on job title and location.
+        for each page we scrap all the job cards shown (usually 15 per page), and for each job card it retrieves its individual link
+        """
+        jobs_Links = []
+        driver = Crawler().get_driver()
+        driver.get(self.url_base)
+        # Find the text box and enter the type of job the user wants to get offers data
+        search = driver.find_element_by_id("text-input-what")
+        search.send_keys(self.job_search)
+        search.send_keys(Keys.RETURN)
+
+        # Get total number of related job offers in all pages.
+        total_job_s = driver.find_element_by_id("searchCountPages").text
+
+        # retrieve the number of total related job offers from string 'for example from Page 1 de 993 emplois we get total_jobs = 993'
+        total_job_s = total_job_s.split()
+        start = total_job_s.index("de")
+        end = total_job_s.index("emplois")
+        total_jobs = int("".join([total_job_s[i] for i in range(start + 1, end)]))
+
+        count_jobs = self.limit  # max 15 jobs per page
+        total_page = int(total_jobs / count_jobs)
+
+        for page in range(0, total_page):
+            if page == self.limit_extract:  # break if page reaches limit set by user
+                break
+            page_url = self.url_base + self.path(pagination=count_jobs * page)
+            driver.get(page_url)
+            sleep(3)
+
+            try:  # get jobCards inside of a page
+                jobCards = WebDriverWait(driver, 10).until(
+                    EC.presence_of_element_located((By.ID, "mosaic-provider-jobcards"))
+                )
+                # get joblink for each jobCard
+                elements = jobCards.find_elements_by_class_name("tapItem")
+                for x in elements:
+
+                    link = x.get_attribute("href")
+                    jobs_Links.append(link)
+
+            finally:
+
+                pass
```

Comment:
* `except` a specific Exception, not a generic Exception. So you have to remove the `finally`
* `jobs_Links` -> `jobs_links` or `jobs_links_list` (snake_case)